This bug has been forwarded upstream
  https://bugs.freedesktop.org/show_bug.cgi?id=6671
Upstream decided to add a de(mac) variant so that keypad
works like on PC.  For now, let's also fix mac keypad.

Index: xkb-data/symbols/macintosh_vndr/de
===================================================================
--- xkb-data.orig/symbols/macintosh_vndr/de
+++ xkb-data/symbols/macintosh_vndr/de
@@ -87,7 +87,7 @@
     key <BKSL> {	[ numbersign,	apostrophe	],
 			[ dead_grave			]	};
 
-    include "keypad(comma)"
+    key <KPDL> {	[ KP_Separator, KP_Delete	]	};
 
     // begin modifier mappings
 //    modifier_map Shift  { Shift_L };
Index: xkb-data/symbols/macintosh_vndr/se
===================================================================
--- xkb-data.orig/symbols/macintosh_vndr/se
+++ xkb-data/symbols/macintosh_vndr/se
@@ -45,7 +45,7 @@
     key <LSGT> {        [            less,         greater      ],
                         [             bar,       brokenbar      ]       };
 
-    include "keypad(comma)"
+    key <KPDL> {	[ KP_Separator, KP_Delete	]	};
 };
 
 
